### PR TITLE
FMS2_io error messages + some clean up

### LINKED ISSUE #739: Update error messages in FMS2_IO
**Is your feature request related to a problem? Please describe.**
Error messages in fms2_io can be improved to include more useful information for debugging

**Describe the solution you'd like**
Improve the content of fms2_io's error messages

**Describe alternatives you've considered**
N/a

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
This PR:

1. Removes `include/get_checksum.inc` in fms2_io because it is not used anywhere in the code
2. Removes the `register_global_attribute` calls in atmosphere_restart_file_test.inc, land_compressed_restart_file_test.inc , land_unstructured_restart_file_test.inc, ocean_restart_file_test.inc, and test_atmosphere_io.F90. `register_global_attribute` is better tested in the test_global_att.F90. 
3. Removes the `SKIP_FMS2_IO_TESTS` flag in the test scripts. 
4. Modifies test_global_att.F90 to test the three supported netcdf file types, and the code only tries to write to `i8_kind` with the netcdf4 file type. If the user is not using the netcdf4 file type and tries to do anything with `i8_kind`, they will get a more helpful error message:
```F90
error(trim(fileobj%path)//": 64 bit integers are only supported with 'netcdf4' file format"//&
                                                       &". Set netcdf_default_format='netcdf4' in the fms2_io namelist OR "//&
                                                       &"add nc_format='netcdf4' to your open_file call")
```
5. Modifies the `check_netcdf_code` subroutine to take in a string, `msg`, that will be appended to the netcdf error code. 
6. Modifies all of the `check_netcdf_code` calls to add a string with the filename, variable, dimension, and/or attribute to make the error message more helfpul instead of just getting netcdf errors like:
```
FATAL: NetCDF: Variable not found
```
7. Cleans up the other error messages to make them more useful. 

Fixes #739 

**How Has This Been Tested?**
`make check` passes with intel20 on skylake and gcc/9.3.0 (netcdf netcdf/4.8.0 and 4.6)
Ran an AM4 experiment with intel18 on gaea and nothing there is broken. 

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: test_fms/fms2_io/test_global_att.F90 [hunk 1]
```diff
@@ -31,14 +33,21 @@ program test_global_att
 integer(kind=i4_kind), dimension(2)    :: buf_i4_kind_1d !< i4_kind 1D buffer
 integer(kind=i8_kind)   :: buf_i8_kind !< i8_kind buffer
 integer(kind=i8_kind), dimension(2)    :: buf_i8_kind_1d !< i8_kind 1D buffer
+character (len = 120), dimension(3) :: my_format !> Array of formats to try.
```
Comment (wrongkindofdoctor) [inline test_fms/fms2_io/test_global_att.F90:36]: !> should be !<

File: test_fms/fms2_io/test_global_att.F90 [hunk 2]
```diff
@@ -48,8 +57,11 @@ program test_global_att
    call register_global_attribute(fileobj, "buf_i4_kind", int(3, kind=i4_kind))
    call register_global_attribute(fileobj, "buf_i4_kind_1d", (/ int(3, kind=i4_kind), int(5, kind=i4_kind) /) )
 
-   call register_global_attribute(fileobj, "buf_i8_kind", int(2, kind=i8_kind))
-   call register_global_attribute(fileobj, "buf_i8_kind_1d", (/ int(2, kind=i8_kind), int(4, kind=i8_kind) /) )
+   !< int8 is only supported with the "netcdf4" type
```
Comment (wrongkindofdoctor) [inline test_fms/fms2_io/test_global_att.F90:60]: !< should be !>

File: test_fms/fms2_io/test_global_att.F90 [hunk 3]
```diff
@@ -69,8 +81,11 @@ program test_global_att
    call get_global_attribute(fileobj, "buf_i4_kind", buf_i4_kind)
    call get_global_attribute(fileobj, "buf_i4_kind_1d", buf_i4_kind_1d)
 
-   call get_global_attribute(fileobj, "buf_i8_kind", buf_i8_kind)
-   call get_global_attribute(fileobj, "buf_i8_kind_1d", buf_i8_kind_1d)
+   !< int8 is only supported with the "netcdf4" type
```
Comment (wrongkindofdoctor) [inline test_fms/fms2_io/test_global_att.F90:84]: !< should be !>

File: test_fms/fms2_io/test_global_att.F90 [hunk 4]
```diff
@@ -80,25 +95,31 @@ program test_global_att
...
-if (buf_i4_kind /= int(3, kind=i4_kind)) call mpp_error(FATAL, "test_global_att: error reading buf_i4_kind")
+if (buf_i4_kind /= int(3, kind=i4_kind)) call mpp_error(FATAL, "test_global_att-"//trim(my_format(i))//": error reading buf_i4_kind")
 if (buf_i4_kind_1d(1) /= int(3, kind=i4_kind) .or. buf_i4_kind_1d(2) /= int(5, kind=i4_kind)) &
-    call mpp_error(FATAL, "test_global_att: error reading buf_i4_kind_1d")
+    call mpp_error(FATAL, "test_global_att-"//trim(my_format(i))//": error reading buf_i4_kind_1d")
 
-if (buf_i8_kind /= int(2, kind=i8_kind)) call mpp_error(FATAL, "test_global_att: error reading buf_i8_kind")
-if (buf_i8_kind_1d(1) /= int(2, kind=i8_kind) .or. buf_i8_kind_1d(2) /= int(4, kind=i8_kind)) &
-    call mpp_error(FATAL, "test_global_att: error reading buf_i8_kind_1d")
+!< int8 is only supported with the "netcdf4" type
```
Comment (wrongkindofdoctor) [inline test_fms/fms2_io/test_global_att.F90:110]: !< should be !>

File: test_fms/fms2_io/test_global_att.F90 [hunk 5]
```diff
@@ -80,25 +95,31 @@ program test_global_att
...
+       & call mpp_error(FATAL, "test_global_att-"//trim(my_format(i))//": error reading buf_i8_kind_1d")
+endif
 
 if (trim(buf_str) /= "some text") then
     print *, "buf_str read in = ", trim(buf_str)
-    call mpp_error(FATAL, "test_global_att: error reading buf_str")
+    call mpp_error(FATAL, "test_global_att-"//trim(my_format(i))//": error reading buf_str")
 endif
+
+enddo !< do i = 1, size(my_format)
```
Comment (wrongkindofdoctor) [inline test_fms/fms2_io/test_global_att.F90:122]: !< should be !>

File: test_fms/fms2_io/test_global_att.F90 [hunk 6]
```diff
@@ -31,14 +33,21 @@ program test_global_att
 integer(kind=i4_kind), dimension(2)    :: buf_i4_kind_1d !< i4_kind 1D buffer
 integer(kind=i8_kind)   :: buf_i8_kind !< i8_kind buffer
 integer(kind=i8_kind), dimension(2)    :: buf_i8_kind_1d !< i8_kind 1D buffer
+character (len = 120), dimension(3) :: my_format !> Array of formats to try.
 
 character(len=20) :: buf_str !< character buffer
+integer :: i !< For Do loop
```
Comment (wrongkindofdoctor) [inline test_fms/fms2_io/test_global_att.F90:39]: !< should be !>

**Review by wrongkindofdoctor — CHANGES_REQUESTED** [review]

**Review by rem1776 — DISMISSED** [review]

**Comment by rem1776** [comment]
@uramirez8707 Just fixed a small conflict in the configure script from merging in #772

**Review by bensonr — DISMISSED** [review]

**Review by wrongkindofdoctor — APPROVED** [review]

**Review by wrongkindofdoctor — DISMISSED** [review]

File: test_fms/fms2_io/test_global_att.F90 (hunk 6 above)
Comment (bensonr) [reply test_fms/fms2_io/test_global_att.F90]: I'm confused here, our style guide shows that the original code is correct "!<"

Comment (wrongkindofdoctor) [reply test_fms/fms2_io/test_global_att.F90:39]: I was going off of the Fortran comment example in the [doxygen manual ](http://cs.swan.ac.uk/~csoliver/ok-sat-library/internet_html/doc/doc/Doxygen/1.7.6.1/html/docblocks.html), but maybe the use of !> or !< doesn't matter for starting lines. I recall having problems rendering the html pages when using !< for subroutine comments, and !> for variables, but maybe this is no longer the case (or I was doing something wrong).

> For Fortran "!>" or "!<" starts a comment and "!!" or "!>" can be used to continuate a one line > comment into a multi-line comment.
>
> Here is an example of a documented Fortran subroutine:
```
  !> Build the restriction matrix for the aggregation 
  !! method.
  !! @param aggr information about the aggregates
  !! @todo Handle special case
  subroutine IntRestBuild(A,aggr,Restrict,A_ghost)
    implicit none
    Type(SpMtx), intent(in) :: A !< our fine level matrix
    Type(Aggrs), intent(in) :: aggr
    Type(SpMtx), intent(out) :: Restrict !< Our restriction matrix

```

Comment (rem1776) [reply test_fms/fms2_io/test_global_att.F90:39]: I believe that is still the case, doxygen's documentation makes them seem interchangeable but when it's actually parsed it tends to miss `!>` if it's at the end of a line so it's best to have `!<` after variables

**Review by bensonr — APPROVED** [review]

**Review by wrongkindofdoctor — APPROVED** [review]

**Review by rem1776 — APPROVED** [review]

**Pull request merged**